### PR TITLE
Tree Preorder Traversal in Dart

### LINKED ISSUE #2550: Tree Preorder Traversal
- [x] C
- [x] C++
- [ ] C# - @sharmakeshav10
- [ ] CoffeeScript
- [x] Dart - @Rukmini-Meda
- [x] Go
- [x] Java
- [ ] JS - @afroz23
- [x] Kotlin - @PrachieNaik
- [x] Php
- [x] Python
- [ ] Readme - @SimranMakhija7
- [x] Ruby
- [ ] Rust
- [ ] TypeScript - @Shr03mink

### PR DESCRIPTION
**Pull request by Rukmini-Meda**
Fixes #2550

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Tree preorder traversal implemented in Dart

#### Languages Used:
- Dart

#### Files Added:
- Tree_Preorder_Traversal.dart

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

**Comment by Rukmini-Meda** [comment]
I have implemented tree preorder traversal in dart and followed the required conventions. Please review the PR so that I can squash commits to merge further. Thank you.

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.dart [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    // 3
+    // 2
+    // 6
+    // 5
+    // 7
+    // 12
+    // 9
+    // 15
+
+}
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.dart:136]: Please add a blank line at end of file.

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.dart [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+        int insert_val = int.parse(input3);
+        tree.insert(root_node, insert_val);
+
+        print("Do you want to insert another number into the tree? (y/n)");
+        var input2 = stdin.readLineSync();
+        choice = input2;
+
+    }
+    
+    // Sample Input tree
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.dart:112]: Please change these multiple single line comments to a multi line comment. Looks neater.

File: Tree_Preorder_Traversal/Tree_Preorder_Traversal.dart [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+    //      / \   / \
+    //     2   6  9  15
+    //        / \
+    //       5   7
+
+    // Calling method to print preorder traversal of the tree.
+    tree.tree_preorder_traversal();
+
+    // Sample Output
+    // The preorder traversal of the tree in the given input is follows:
```
Comment (singh-shreya6) [inline Tree_Preorder_Traversal/Tree_Preorder_Traversal.dart:125]: Please remove these single line comments and make it a multi line comment.

**Comment by Rukmini-Meda** [comment]
I have made the requested changes and squashed the commits. Please review and merge. Thank you.

**Pull request merged**